crates/aranya-policy-compiler/src/tests.rs

```diff
@@ -2124,6 +2130,116 @@ fn test_validate_return() {
     }
 }
 
+#[test]
+fn test_validate_publish() {
+    let concat = |text| {
+        let base = r#"
+            command Foo {
+                fields {
+                    a int
+                }
+                seal { return todo() }
+                open { return todo() }
+                policy {
+                    finish {}
+                }
+                recall {
+                    finish {}
+                }
+            }
+        "#;
+        format!("{base}{text}")
+    };
+
+    let valid = [
+        concat(
+            r#"
+            action a() {
+                publish Foo { a: 0 } // ok
+            }
+        "#,
+        ),
+        concat(
+            r#"
+            action b() {
+                if true {}
+
+                publish Foo { a: 0 } // ok
+            }
+        "#,
+        ),
+        concat(
+            r#"
+            action c() {
+                if true {}
+                else {
+                    publish Foo { a: 0 }
+                }
+                publish Foo { a: 1 }
+            }
+        "#,
+        ),
+        concat(
+            r#"
+            action d() {
+                if true {
+                    publish Foo { a: 0 }
+                }
+                else {
+                    publish Foo { a: 1 }
+                }
+            }
+        "#,
+        ),
+        concat(
+            r#"
+            action e() {
+                let n = 0
+                match n {
+                    0 => { publish Foo { a: 0 } }
+                    _ => { publish Foo { a: 1 } }
+                }
+            }
```

Comment:
Rewriting this as:

```suggestion
            action e(n int) {
                match n {
                    0 => { publish Foo { a: 0 } }
                    _ => { publish Foo { a: 1 } }
                }
            }
```

causes the ValueAnalyzer to fail 😕

Reply:
Looks like we shouldn't add the action args to `predefined`, like we don't for functions.

Reply:
Ah, that makes sense. Removing that causes the 2 invalid cases I wrote to now become valid. Looking into that now...